### DISCUSSION TITLE
Metadaten ohne Typ: Was kann das verursachen?

### DISCUSSION THREAD

**Discussion by matthias-ronge** [kitodo/kitodo-production#5838] [Q&A]
Bei einer Einrichtung scheiterte der Export. Nach der Untersuchung des Falles stellte ich fest, dass die Metadatensätze des fehlgeschlagenen Vorgangs Metadaten mit einem leeren Typ enthielten (`<kitodo:metadata name="">value</kitodo:metadata>`). (Im konkreten Fall ist XSLT fehlgeschlagen, weil es `<kitodo:metadata name="key">value</kitodo:metadata>` nach`<institut:key>value</institut:key>` abbildet. Bei einem leeren `name` führt dies zu ungültigem XML wie `<institut:>value</institut:>`.)

Ich frage mich, **wie können Metadaten ohne Typ in den Vorgang gelangt sein?** Die Anwender wussten es auch nicht mehr, da der Vorgang schon recht alt war. Irgendwelche Ideen, wo dies möglich ist, oder war? Wenn wir das irgendwie reproduzieren können, sollten wir es abfangen.

**Comment by henning-gerhardt** [comment]
Was ist mit "der Vorgang schon recht alt war" gemeint? Ist dies ein Vorgang aus Kitodo.Production 2.x und wurde via XSLT migriert? Wenn ja, dann könnte es an der Migration aus dem alten Kitodo.Production 2.x ins neue Kitodo.Production 3.x Format liegen. Ggf. existiert noch die alte 2.x Metadaten Datei, so dass man darüber heraus bekommt, wie der Typ des Metadatums war.

Unabhängig davon sollte bei der Metadaten-Validierung (Metadaten-Editor als auch Export) es einen Fehler geben, wenn ein Metadatum gefunden wird, dass keinen Typ hat. Ob dieses Metadatum einen Wert enthält, ist an der Stelle nachrangig.

**Comment by andre-hohmann** [comment]
Tritt dieser Fehler häufig auf?
Wenn nicht, stelle ich mir die Untersuchung der Ursache schwer vor und ich würde mich fragen, ob es sich lohnt, diese durchzuführen. 
Unter Umständen war es eine temporäre fehlerhafte Konfiguration vor mehreren Jahren.

Ich stimme Henning zu, dass es notwendig ist, in welchem System und in welchem Kontext die Transformation durchgeführt wurde. 
Mir ist nicht bekannt, dass dieses Verhalten in der SLUB Dresden aufgetreten ist.

**Comment by matthias-ronge** [comment]
Es handelt sich tatsächlich um ein migriertes System, der Vorgang wurde migriert und möglicherweise ist es ein Überbleibsel. Der Vorgang aus 2.x ist nicht mehr vorhanden. Ich dachte nur, vielleicht hat jemand eine Idee, wie es dazu kommen könnte. Ich fühle mich nicht wohl, wenn so etwas öfter passieren sollte, aber wenn es sonst niemand gesehen hat, sollte es in Ordnung sein.
